### ISSUE TITLE
#@> #@$@>  #@,@> special code

### ISSUE THREAD

**Issue by HenryHRich** [jsoftware/jsource#90]

**Comment by ramonasuncion** [comment]
I'll take a look at this if you aren't already. I've been reading through `ca.c` and special codes

**Comment by ramonasuncion** [comment]
Yeah... I’ll leave this to someone more familiar with the codebase and J’s parsing rules. 😅

**Comment by HenryHRich** [comment]
No, man, give it a try!  It's a useful addition & I can easily explain how to do it.  If you can build the code and run the testcases you are more than halfway there.

**Comment by ramonasuncion** [comment]
I'll give it another shot. I do have everything setup.

Yesterday I tried using the existing #@> optimization:

https://github.com/jsoftware/jsource/blob/01eb9b6951ee964445153a1f2dfb5694524688d4/jsrc/ca.c#L194-L199

 But I got stuck figuring out how J groups the extra @ in #@$@> and #@,@>.

**Comment by HenryHRich** [comment]
Conjunctions have long left scope, that is #@$@> groups as (#@$)@> .  
This is implied by the parsing table but you have to dig pretty deep to 
see why (it's because line 4 doesn't match C in the first word).

The advantage of this grouping is that /execution/ is right to left: > 
then $ then # .

You are looking at the right code: you need a function that works like 
jttallyatopopen but returns AR or AN; then you need to modify ca.c to 
call that function at the appropriate time.

hhr

**Comment by ramonasuncion** [comment]
I worked on the first one to get some feedback.

**Comment by HenryHRich** [comment]
Implemented in beta7
